src/RadixDlt.NetworkGateway.Abstractions/CoreCommunications/TransactionSubmitter.cs

```diff
@@ -0,0 +1,240 @@
+
+    public record IntentAlreadyCommitted(CommittedIntentMetadata CommittedAsDetails) : NodeSubmissionResult;
+
+    /// <summary>
+    /// If the transaction was accepted into the mempool.
+    /// This suggests the node believe the transaction could be committed.
+    /// </summary>
+    /// <param name="IsAlreadyThere">True if the transaction was already in the given node's mempool.</param>
+    public record AcceptedIntoMempool(bool IsAlreadyThere) : NodeSubmissionResult;
+
+    public record Rejection(TransactionSubmitRejectedErrorDetails Details) : NodeSubmissionResult;
+
+    public record MempoolPriorityThresholdNotMet(TransactionSubmitPriorityThresholdNotMetErrorDetails Details) : NodeSubmissionResult;
+
+    public record OtherSubmissionError(Exception Exception, string ErrorTitle) : NodeSubmissionResult;
+
+    private NodeSubmissionResult()
+    {
+    }
+
+    public bool ShouldStopSubmittingPermanently()
+    {
+        return this switch
+        {
+            IntentAlreadyCommitted => true,
+            AcceptedIntoMempool => false,
+            Rejection rejection => rejection.Details.IsPayloadRejectionPermanent,
+            MempoolPriorityThresholdNotMet => false,
+            _ => throw new ArgumentOutOfRangeException(),
+        };
+    }
+
+    public PendingTransactionPayloadLedgerStatus PayloadStatus()
+    {
+        return this switch
+        {
+            IntentAlreadyCommitted commit => commit.CommittedAsDetails.IsSameTransaction
+                ? PendingTransactionPayloadLedgerStatus.CommitPendingOutcomeUnknown
+                : PendingTransactionPayloadLedgerStatus.CommitOfOtherPayloadForIntentPendingOutcomeUnknown,
+            AcceptedIntoMempool => PendingTransactionPayloadLedgerStatus.TransientlyAccepted,
+            Rejection rejection => rejection.Details.IsPayloadRejectionPermanent
+                ? PendingTransactionPayloadLedgerStatus.PermanentlyRejected
+                : PendingTransactionPayloadLedgerStatus.TransientlyRejected,
+            _ => PendingTransactionPayloadLedgerStatus.Unknown,
+        };
+    }
+
+    public PendingTransactionIntentLedgerStatus IntentStatus()
+    {
+        return this switch
+        {
+            IntentAlreadyCommitted => PendingTransactionIntentLedgerStatus.CommitPendingOutcomeUnknown,
+            AcceptedIntoMempool => PendingTransactionIntentLedgerStatus.PossibleToCommit,
+            Rejection rejection => rejection.Details.IsIntentRejectionPermanent
+                ? PendingTransactionIntentLedgerStatus.PermanentRejection
+                : PendingTransactionIntentLedgerStatus.LikelyButNotCertainRejection,
+            _ => PendingTransactionIntentLedgerStatus.Unknown,
+        };
+    }
+
+    public string? GetExecutionFailureReason()
+    {
+        return this switch
+        {
+            Rejection rejection => rejection.Details.ErrorMessage,
+            _ => null,
+        };
+    }
+
+    public string? GetApiSubmitErrorMessage()
```

Comment:
IMO it would be better to throw exception here if we're dealing with successful result. Client to this piece of code should not call this method on success.

i.e. avoid nulls if possible.

Reply:
Maybe? Depends how this is used. Will review.